lib/chef/provider/git.rb

```diff
@@ -157,8 +157,8 @@ def checkout
 
         converge_by("checkout ref #{sha_ref} branch #{@new_resource.revision}") do
           # checkout into a local branch rather than a detached HEAD
-          shell_out!("git branch -f #{@new_resource.checkout_branch} #{sha_ref}", run_options(:cwd => @new_resource.destination))
-          shell_out!("git checkout #{@new_resource.checkout_branch}", run_options(:cwd => @new_resource.destination))
+          git(["branch", "-f", @new_resource.checkout_branch, sha_ref], { cwd: cwd })
+          git(["checkout", @new_resource.checkout_branch], { cwd: cwd })
```

Comment:
...and that'll let you write this as `git("checkout", @new_resource.checkout_branch, cwd: cwd)` along with removing the rest of the array and hash decoration